src/config/cli_tools.cpp

```diff
@@ -0,0 +1,143 @@
+#include "cli_tools.hpp"
+
+#include <pwd.h>
+
+#include <filesystem>
+
+#include "cli_config.hpp"
+#include "devnet_config.hpp"
+#include "testnet_config.hpp"
+
+using namespace std;
+using namespace dev;
+namespace fs = std::filesystem;
+
+namespace taraxa {
+
+void CliTools::generateConfig(const std::string &config, const std::string &data_dir, bool boot_node, int network_id) {
+  Json::Value conf;
+  switch (network_id) {
+    case CliConfig::Testnet:
+      conf = readJsonFromString(testnet_json);
+      break;
+    case CliConfig::Devnet:
+      conf = readJsonFromString(devnet_json);
+      break;
+    default:
+      throw invalid_argument("Unsupported network ID");
+  }
+  conf["db_path"] = data_dir;
+  conf["network_is_boot_node"] = boot_node;
+
+  writeJsonToFile(config, conf);
+}
+
+void CliTools::generateWallet(const std::string &wallet) {
+  // Wallet
+  auto account = dev::KeyPair::create();
+  auto [pk, sk] = taraxa::vrf_wrapper::getVrfKeyPair();
+  auto account_json = createWalletJson(account, sk, pk);
+
+  // Create account file
+  writeJsonToFile(wallet, account_json);
+}
+
+void CliTools::generateAccount(dev::KeyPair account) {
+  cout << "secret : " << toHex(account.secret().ref()) << endl;
+  ;
+  cout << "public : " << account.pub().toString() << endl;
+  ;
+  cout << "address : " << account.address().toString() << endl;
+  ;
+}
+
+void CliTools::generateAccount() { generateAccount(dev::KeyPair::create()); }
+
+void CliTools::generateAccountFromKey(string key) {
+  auto sk = dev::Secret(key, dev::Secret::ConstructFromStringType::FromHex);
+  auto account = dev::KeyPair(sk);
+  generateAccount(account);
+}
+
+void CliTools::generateVrf(taraxa::vrf_wrapper::vrf_sk_t sk, taraxa::vrf_wrapper::vrf_pk_t pk) {
+  cout << "vrf_secret : " << sk.toString() << endl;
+  cout << "vrf_public : " << pk.toString() << endl;
+}
+
+void CliTools::generateVrf() {
+  auto [pk, sk] = taraxa::vrf_wrapper::getVrfKeyPair();
+  generateVrf(sk, pk);
+}
+
+void CliTools::generateVrfFromKey(string key) {
+  auto sk = taraxa::vrf_wrapper::vrf_sk_t(key);
+  auto pk = taraxa::vrf_wrapper::getVrfPublicKey(sk);
+  generateVrf(sk, pk);
+}
+
+Json::Value CliTools::createWalletJson(dev::KeyPair account, taraxa::vrf_wrapper::vrf_sk_t sk,
+                                       taraxa::vrf_wrapper::vrf_pk_t pk) {
+  Json::Value json(Json::objectValue);
+  json["node_secret"] = toHex(account.secret().ref());
+  json["node_public"] = account.pub().toString();
+  json["node_address"] = account.address().toString();
+  json["vrf_secret"] = sk.toString();
+  json["vrf_public"] = pk.toString();
+  return json;
+}
+
+Json::Value CliTools::getAccountSecretKey(const string &wallet) { return readJsonFromFile(wallet)["node_secret"]; }
+
+Json::Value CliTools::getVrfSecretKey(const string &wallet) { return readJsonFromFile(wallet)["vrf_secret"]; }
+
+string CliTools::getHomeDir() { return string(getpwuid(getuid())->pw_dir); }
```

Comment:
I guess for this use case it is ok as there is not many calls to these functions but in general might be a better idea to create all these variables in constructor and then just return const std::string &